napari/layers/image/experimental/octree.py

```diff
@@ -67,240 +74,285 @@ def _create_tiles(array: np.ndarray, tile_size: int) -> np.ndarray:
     return tiles
 
 
-def _create_downsampled_tile(
-    ul: np.ndarray, ur: np.ndarray, ll: np.ndarray, lr: np.ndarray
-) -> np.ndarray:
+def _get_tile(tiles: TileArray, row, col):
+    try:
+        return tiles[row][col]
+    except IndexError:
+        return None
+
+
+def _none(items):
+    return all(x is None for x in items)
+
+
+def _combine_tiles(*tiles: np.ndarray) -> np.ndarray:
+    """Combine 1-4 tiles into a single tile.
+
+    Parameters
+    ----------
+    tiles
+        The 4 child tiles, some might be None.
```

Comment:
I take it right now this is `Quadtree`, but will then become `Octree` soon. That's fine, but I just want to check I am not missing something

Reply:
It started as a real Octree with 8 children per node, but we just happened to only use 4 children because we are only doing 2D. But then I realized that a "region quadtree" or "region octree" that holds images doesn't really need nodes and children. The Raveler one didn't have them.

So now it is really a region quadtree. Where you have just the N levels and each level is just a 2D array. And there are no nodes and no child pointers/references. But you can go from parent to child and back just be halving or doubling the indexes.

But everything should be the same with 3D/octree but just extended to 3D. At that point we'll probably just have one set of octree/3D code and just use it in 2D for 2D.

But yes right now it's really a region quadtree that aspires to be an octree.